### PR TITLE
API call for upgrading group configuration

### PR DESCRIPTION
**Pull request by druizz90**
Hi team,

This PR  is for issue #245. I added a new call for upgrading group configurations. This is an example of this new call:

```bash
curl -u foo:bar -X POST -H 'Content-type: application/xml' -d @agent.conf.xml "http://localhost:55000/agents/groups/dmz/configuration?pretty"
```

With the parameter `@agent.conf.xml` the content of this file will upgrade the group configuration (`dmz` in this example).

This feature is only available for upgrading `agent.conf` file, but this will be available for more files in the future.

It was necessary to change some permissions for execute our binary `verify-agent-conf` in `install_api.sh` script.

Best regards,

Demetrio.

### CODE REVIEW

File: install_api.sh [hunk 1]
```diff
@@ -333,6 +333,13 @@ setup_api_permissions () {
 
     # API users permissions
     exec_cmd "chmod 660 $API_PATH/configuration/auth/user"
+
+    # permissions for execute verify-agent-conf binary
```
Comment (jesuslinares) [inline install_api.sh]: This must be done in the makefile / install.sh of Wazuh.

Comment (druizz90) [reply install_api.sh]: ok!

**Comment by mgmacias95** [comment]
Hello @DaveVG1,

Thank you so much for your feedback. Can you give us more detail about the last one? I have tried to reproduce the issue but, in my case, the agent.conf file isn't overwritten when the file size is higher than 1MB. Here's a snippet showing it:
```shellsession
# ls -lah /var/ossec/etc/shared/webserver/agent.conf
-rw-r--r-- 1 ossec ossec 47 Jan  8 11:01 /var/ossec/etc/shared/webserver/agent.conf
# ls -lah agent.conf.xml
-rw-r--r-- 1 root root 47M Jan  8 11:13 agent.conf.xml
# curl -u foo:bar -X POST -H 'Content-type: application/xml' -d @agent.conf.xml "http://localhost:55000/agents/groups/webserver/configuration?pretty"
{"error":"701","message":"Size of XML file is too long"}
# ls -lah agent.conf.xml
-rw-r--r-- 1 root root 47M Jan  8 11:13 agent.conf.xml
# ls -lah /var/ossec/etc/shared/webserver/agent.conf
-rw-r--r-- 1 ossec ossec 47 Jan  8 11:01 /var/ossec/etc/shared/webserver/agent.conf
```

Best regards,
Marta

**Comment by mgmacias95** [comment]
Hello team,

I have been improving code in this PR and https://github.com/wazuh/wazuh/pull/2155:

* `POST/agents/groups/:group_id/files/:file_name` API call:
    ```javascript
    # curl -u foo:bar -X POST -H 'Content-type: application/xml' -d @agent.conf.xml "http://localhost:55000/agents/groups/webserver/files/agent.conf?pretty"
    {
       "error": 0,
       "data": "Agent configuration was updated successfully"
    }
    ```
* Only `agent.conf` file is allowed:
    ```json
    # curl -u foo:bar -X POST -H 'Content-type: application/xml' -d @agent.conf.xml "http://localhost:55000/agents/groups/webserver/files/aaaaaa?pretty"
    {
       "error": 1111,
       "message": "Remote group file updates are only available in 'agent.conf' file"
    }
    ````
* Raise exception if an empty file is received:
    ```json
    # curl -u foo:bar -X POST -H 'Content-type: application/xml' -d @agent.conf "http://localhost:55000/agents/groups/webserver/files/agent.conf?pretty"
    Warning: Couldn't read data from file "agent.conf", this makes an empty POST.
    {
       "error": 1112,
       "message": "Empty files aren't supported"
    }
    ```
* Raise exception if XML isn't correct.
    ```json
    # curl -u foo:bar -X POST -H 'Content-type: application/xml' -d @agent.conf.xml "http://localhost:55000/agents/groups/webserver/files/agent.conf?pretty"
    {
       "error": 1113,
       "message": "XML syntax error: mismatched tag: line 1, column 30"
    }
    ```
* Raise exception if agent configuration is not validated by `verify-agent-conf` binary:
    ```json
    # curl -u foo:bar -X POST -H 'Content-type: application/xml' -d @agent.conf.xml "http://localhost:55000/agents/groups/webserver/files/agent.conf?pretty"
    {
       "error": 1114,
       "message": "Wazuh syntax error: Invalid element in the configuration: 'agent_confi'. Syscheck remote configuration in '/var/ossec/tmp/api_tmp_file_2019-01-08-01-1546960883.xml' is corrupted."
    }
    ```
* Raise exception if received agent.conf size is higher than 1MB:
     ```json
     # curl -u foo:bar -X POST -H 'Content-type: application/xml' -d @agent.conf.xml "http://localhost:55000/agents/groups/webserver/files/agent.conf?pretty"
    {"error":"701","message":"Size of XML file is too long"}
    ```

Best regards,
Marta

**Pull request merged**